src/promptflow/promptflow/_sdk/_submitter/utils.py

```diff
@@ -198,7 +198,20 @@ def resolve_connections(flow: Flow, client=None, connections_to_ignore=None) ->
         )
 
     @staticmethod
-    def resolve_connection_names_from_tool_meta(tools_meta: dict):
+    def resolve_connection_names_from_tool_meta(tools_meta: dict, flow_dag: dict,):
+        connection_names = set({})
+        tool_names = set({})
+        if tools_meta:
+            packages = tools_meta.get("package", {})
+            for key, pkg in packages.items():
+                inputs = pkg.get("inputs", {})
+                if "connection" in inputs and inputs["connection"].get("type", []) == ["object"]:
```

Comment:
I'm afraid that you can't assume that all connection inputs are named `connection`